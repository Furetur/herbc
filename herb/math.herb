fn fact(n: int) -> int {
    if n <= 1 {
        return 1;
    }
    return n * fact(n - 1);
}

fn is_prime(x: int) -> bool {
    var i = 2;

    while i * i <= x {
        if x % i == 0 {
            return false;
        }
        i = i + 1;
    }
    return true;
}

fn fib(n: int) -> int {
    var a = 1;
    var b = 1;

    var i = 2;

    while i < 10 {
        var next = a + b;
        a = b;
        b = next;
        i = i + 1;
    }
    return next;
}

fn phi(x: int) -> int {
    var n = x;
	var result = n;

	var i = 2;
	while i * i <= n {
	    if n % i == 0 {
	        while n % i == 0 {
	            n = n / i;
	        }
	        result = result - result / i;
	    }
	    i = i + 1;
	}
	if n > 1 {
	    result = result - result / n;
	}
	return result;
}
