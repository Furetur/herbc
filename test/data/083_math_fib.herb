import herb.math;

entrypoint {
    print(math.fib(30));
    print(" ");
    print(math.fib(40));
}