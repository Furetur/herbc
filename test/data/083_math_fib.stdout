832040 102334155